src/foam/nanos/crunch/CapabilityAvailabilityDAO.js

```diff
@@ -96,6 +97,10 @@ foam.CLASS({
         { type: 'Capability', name: 'capability' }
       ],
       javaCode: `
+        // TODO: Rules calling capabilityDAO under the system context are
+        //   breaking availability logic. This needs proper investigation.
+        if ( ((Subject) x.get("subject")).getUser().isAdmin() ) return true;
```

Comment:
IsAdmin - This property raises a red flag for me - why and what defines an admin on a nanos system? Our authorization system is primarily based on permissions and user owned objects defining those permissions. Why are we not using permissions to resolve this?

Reply:
Glad you brought this up in review, I wanted to bring attention to it on Slack but I forgot.

> why and what defines an admin on a nanos system?
`IsAdmin` does - but in the documentation for that method I explicitly wrote that we shouldn't use it for this purpose because of exactly what you said: that's what the permission system is for.

The call to `capabilityDAO` from a system context is the real problem. `CapabilityAvailabilityDAO` is nonsense for any operation you'd expect to do under a system context, since its behaviour is entirely reliant on a meaningful `subject` in the context.

We could also create a permission for capability availability and explicitly check for that; my worry is if we did that it would encourage calling `capabilityDAO` from a system context, which may cause us other problems in the future.

Currently I'm working on a `PreventSystemDAO` that we can decorate the served `capabilityDAO` with to warn about or prevent this behaviour.

Reply:
@zahnno I've confirmed that this check isn't needed anymore